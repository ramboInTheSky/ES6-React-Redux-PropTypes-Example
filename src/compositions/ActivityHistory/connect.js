import { connect } from 'react-redux';
import path from 'ramda/src/path';

import { getActivityHistory, invalidateActivityHistory } from '../../ducks/activities';

export const mapStateToProps = state => {
  const {
    activityHistory: { heading, labels },
  } = state.dictionary;

  return {
    activityHistory: path(['activities', 'history'], state),
    heading,
    labels,
  };
};

export const mergeProps = (
  { arrears, tenants, vulnerabilityFlag, ...stateProps },
  { dispatch },
  ownProps
) => {
  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);

  return {
    ...stateProps,
    arrearsId,
    invalidateActivityHistory: () => dispatch(invalidateActivityHistory()),
    getActivityHistory: () => dispatch(getActivityHistory(arrearsId)),
    ...ownProps,
  };
};

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
