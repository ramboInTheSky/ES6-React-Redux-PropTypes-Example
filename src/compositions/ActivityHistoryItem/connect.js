import { connect } from 'react-redux';
import path from 'ramda/src/path';

import activityNavigation from '../../util/activityNavigation';
import getHelpers from '../../util/stateHelpers';

import { updateRibbon } from '../../ducks/ribbon';
import { getActivity, getActivityHistory, invalidateActivityDetail } from '../../ducks/activities';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);

  const {
    activityHistory: { itemHeading: heading, labels, values, attachments: attachmentsHeading },
    arrearsDetails: { heading: arrearsDetailsHeading },
    genericErrorText,
    paymentPlan: { installmentArrangementFormat },
  } = state.dictionary;

  const referralLabels = path(['dictionary', 'referral'], state);
  const pauseLabels = path(['dictionary', 'pause'], state);
  const paymentLabels = path(['dictionary', 'payment', 'labels'], state);
  const paymentPlanLabels = path(['dictionary', 'paymentPlan', 'labels'], state);
  const legalReferralLabels = path(['dictionary', 'legalReferral'], state);

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const itemId = path(['match', 'params', 'itemId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);
  const goToHistoryItem = historyId =>
    ownProps.history.push(`/arrears-details/${arrearsId}/history/${historyId}`);
  const activities = get(['activities', 'history'], []);
  const attachments = get(['activities', 'attachments']);
  const attachmentsLoading = get(['activities', 'attachmentsLoading'], []);
  const attachmentsError = get(['activities', 'attachmentsError'], []);
  const activityHistoryNavigation = activityNavigation(activities, itemId);

  return {
    activities,
    activityHistoryNavigation,
    genericErrorText,
    attachmentsHeading,
    activityDetail: get(['activities', 'detail']),
    activityError: get(['activities', 'error']),
    activityLoading: get(['activities', 'loading']),
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    canGoNext: !activityHistoryNavigation.isLast(),
    canGoPrevious: !activityHistoryNavigation.isFirst(),
    heading,
    goToHistoryItem,
    itemId,
    labels,
    redirectToDetailsPage,
    values,
    referralLabels,
    pauseLabels,
    paymentLabels,
    paymentPlanLabels: { installmentArrangementFormat, ...paymentPlanLabels },
    legalReferralLabels,
    attachments,
    attachmentsError,
    attachmentsLoading,
  };
};

export const mergeProps = (
  {
    activities,
    activityDetail,
    activityHistoryNavigation,
    arrearsId,
    breadcrumb,
    heading,
    goToHistoryItem,
    itemId,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  activities,
  itemId,
  activityDetail,
  getActivityById: id => {
    const activityHistoryItem = activities.find(activity => activity.id === id);
    dispatch(getActivity(activityHistoryItem)).catch(() => {});
  },
  getActivityHistory: () => {
    dispatch(getActivityHistory(arrearsId));
  },
  invalidateActivityDetail: () => dispatch(invalidateActivityDetail()),
  onBack: () => redirectToDetailsPage(),
  onNext: () => {
    const nextHistoryId = activityHistoryNavigation.getNextHistoryId();
    if (nextHistoryId) {
      goToHistoryItem(nextHistoryId);
    }
  },
  onPrevious: () => {
    const prevHistoryId = activityHistoryNavigation.getPreviousHistoryId();
    if (prevHistoryId) {
      goToHistoryItem(prevHistoryId);
    }
  },
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
