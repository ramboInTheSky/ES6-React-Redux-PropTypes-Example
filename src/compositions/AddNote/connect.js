import { connect } from 'react-redux';
import path from 'ramda/src/path';

import getHelpers from '../../util/stateHelpers';
import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import { addNoteToCase, clearNoteError } from '../../ducks/notes';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    addNote: { addNote, formError, notification, ...text },
    arrearsDetails: { heading: arrearsDetailsHeading },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const returnUrl = `/arrears-details/${arrearsId}`;
  const redirectToDetailsPage = () => ownProps.history.push(returnUrl);

  const redirectToAttachmentsPage = id =>
    ownProps.history.push(`${returnUrl}/note/${id}/attachments?redirectTo=${returnUrl}`);

  return {
    addNote,
    arrearsId,
    breadcrumb: getBreadcrumbs(
      addNote,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    error: get(['notes', 'error']),
    isLoading: get(['notes', 'loading']),
    formError,
    heading: addNote,
    notification,
    redirectToDetailsPage,
    redirectToAttachmentsPage,
    ...text,
  };
};

export const mergeProps = (
  {
    arrearsId,
    breadcrumb,
    heading,
    notification,
    redirectToAttachmentsPage,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  arrearsId,
  clearError: () => dispatch(clearNoteError()),
  onBack: redirectToDetailsPage,
  onSubmit: (note, wantsToAttachFiles) =>
    dispatch(addNoteToCase(arrearsId, note))
      .then(id => {
        dispatch(
          addNotification({
            dataBddPrefix: 'addNote',
            lines: [{ text: notification.line1 }],
            icon: 'success',
            notificationType: 'confirmation',
            title: notification.title,
          })
        );

        if (wantsToAttachFiles) {
          redirectToAttachmentsPage(id);
        } else {
          redirectToDetailsPage();
        }
      })
      .catch(() => {}),
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
