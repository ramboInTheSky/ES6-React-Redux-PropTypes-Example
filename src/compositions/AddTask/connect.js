import { connect } from 'react-redux';
import path from 'ramda/src/path';
import { formatting } from 'nhh-styles';
import getHelpers from '../../util/stateHelpers';
import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import { clearTaskError, taskError, addTask as AddTaskItem } from '../../ducks/tasks';
import { clearSearch, search } from '../../ducks/housingofficersearch';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    addTask,
    addTask: { addTask: heading, notification, formError, serverError },
    arrearsDetails: { heading: arrearsDetailsHeading },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  return {
    addTask,
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    loading: get(['tasks', 'loading']),
    housingOfficerSearchResults: get(['housingofficersearch', 'searchResults']) || [],
    error: get(['tasks', 'error']),
    userId: get(['user', 'profile', 'id']),
    loggedInUserId: get(['user', 'profile', 'id']),
    userFullName: get(['user', 'profile', 'fullname']),
    notification,
    heading,
    redirectToDetailsPage,
    errorMessage: get(['tasks', 'isFormError']) ? formError : serverError,
  };
};

export const mergeProps = (
  {
    arrearsId,
    breadcrumb,
    heading,
    housingOfficerSearchResults,
    notification,
    redirectToDetailsPage,
    userId,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  arrearsId,
  housingOfficerSearchResults,
  clearUserSearch: () => dispatch(clearSearch()),
  onBack: () => redirectToDetailsPage(),
  onDisplayFormError: () => dispatch(taskError(true)),
  onSubmit: ({ title, description, date, ownerId }) => {
    dispatch(AddTaskItem(arrearsId, title, description, date, ownerId, userId))
      .then(() => {
        dispatch(
          addNotification({
            dataBddPrefix: 'addTask',
            lines: [{ text: `${title} ${formatting.formatDate(date)}` }],
            icon: 'success',
            notificationType: 'confirmation',
            title: notification.title,
          })
        );
        redirectToDetailsPage();
      })
      .catch(() => {});
  },
  onSearch: term => {
    if (term && term.length >= 3) {
      dispatch(search(term));
    } else if (term.length === 0 && !!housingOfficerSearchResults.length) {
      dispatch(clearSearch());
    }
  },
  onUnmount: () => {
    dispatch(clearTaskError());
    dispatch(clearSearch());
  },
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  // TODO: remove this next line. By the looks of it is not needed anymore. However, I'm not 100% sure.
  // So I'm just leaving it here in case this later becomes a bug, in QA.
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
