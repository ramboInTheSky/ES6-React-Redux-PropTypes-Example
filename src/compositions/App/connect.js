import compose from 'ramda/src/compose';
import pickAll from 'ramda/src/pickAll';
import { connect } from 'react-redux';

import { displayLogin, logout } from '../../ducks/user';
import { setEntryPoint } from '../../ducks/navigation';
import { hideModal } from '../../ducks/modal';

export const mapStateToProps = pickAll(['user', 'ribbon', 'modal']);

export const mergeProps = (stateProps, { dispatch }, ownProps) => ({
  ...stateProps,
  login: () => dispatch(displayLogin()),
  logout: () => dispatch(logout()),
  modal: {
    ...stateProps.modal,
    onClose: compose(
      dispatch,
      hideModal
    ),
  },
  setEntryPoint: path => dispatch(setEntryPoint(path)),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
