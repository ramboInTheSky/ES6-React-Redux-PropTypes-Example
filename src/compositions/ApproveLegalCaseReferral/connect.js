import { connect } from 'react-redux';
import path from 'ramda/src/path';

import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import {
  cancelLegalReferral,
  getLegalReferral,
  setLegalReferralApproved,
} from '../../ducks/legalReferral';
import getHelpers from '../../util/stateHelpers';
import downloadFile from '../../util/downloadFile';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    arrearsDetails: { heading: arrearsDetailsHeading },
    approvelegalCaseReferral: { errorText, heading, labels, notification, subtitle },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const submissionId = path(['match', 'params', 'submissionId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  return {
    activePause: get(['pause', 'activePause']),
    arrearsId,
    breadcrumb: getBreadcrumbs(
      heading,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    errorText,
    formError: get(['pause', 'error']),
    heading,
    isManager: get(['user', 'userType']) === 'manager',
    labels,
    loading: get(['legalReferral', 'loading']),
    notification,
    referralPack: get(['legalReferral', 'detail', 'referralPack']),
    submissionId,
    subtitle,
    redirectToDetailsPage,
  };
};

export const mergeProps = (
  {
    breadcrumb,
    heading,
    notification: { approvedTitle, cancelledTitle },
    submissionId,
    redirectToDetailsPage,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  getLegalReferral: () => dispatch(getLegalReferral(submissionId)),
  heading,
  onBack: () => redirectToDetailsPage(),
  onSubmit: noteText =>
    dispatch(
      noteText
        ? cancelLegalReferral(submissionId, { noteText })
        : setLegalReferralApproved(submissionId)
    )
      .then(() => {
        redirectToDetailsPage();
        dispatch(
          addNotification({
            dataBddPrefix: 'approveLegalReferral',
            icon: 'success',
            notificationType: 'confirmation',
            title: noteText ? cancelledTitle : approvedTitle,
          })
        );
      })
      .catch(() => {}),
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  downloadFile: (uri, filename) => downloadFile(uri, filename),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
