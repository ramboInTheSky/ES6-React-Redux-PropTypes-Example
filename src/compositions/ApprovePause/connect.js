import { connect } from 'react-redux';
import path from 'ramda/src/path';
import { formatting } from 'nhh-styles';
import format from 'string-format';

import { addNotification } from '../../ducks/notifications';
import { updateRibbon } from '../../ducks/ribbon';
import { approvePause, getPause, getPauseRules } from '../../ducks/pause';
import getHelpers from '../../util/stateHelpers';

export const mapStateToProps = (state, ownProps) => {
  const { get, getBreadcrumbs } = getHelpers(state);
  const {
    approvePause: { errorText, labels, notification, pauseRequest, pauseRequestDetails },
    arrearsDetails: { heading: arrearsDetailsHeading },
  } = state.dictionary;

  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);
  const pauseId = path(['match', 'params', 'pauseId'], ownProps);
  const redirectToDetailsPage = () => ownProps.history.push(`/arrears-details/${arrearsId}`);

  return {
    activePause: get(['pause', 'activePause']),
    arrearsId,
    breadcrumb: getBreadcrumbs(
      pauseRequest,
      {},
      {
        label: arrearsDetailsHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    errorText,
    formError: get(['pause', 'error']),
    heading: pauseRequestDetails,
    labels,
    loading: get(['pause', 'loading']),
    notification,
    pauseId,
    redirectToDetailsPage,
    rules: get(['pause', 'pauseRules']),
    userId: get(['user', 'profile', 'id']),
  };
};

export const mergeProps = (
  {
    activePause,
    arrearsId,
    breadcrumb,
    heading,
    notification: { approvedTitle, cancelledTitle, line1, line2 },
    pauseId,
    redirectToDetailsPage,
    userId,
    ...stateProps
  },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  activePause,
  getPause: () =>
    dispatch(getPause(arrearsId, pauseId)).then(() => dispatch(getPauseRules(arrearsId, userId))),
  onBack: () => redirectToDetailsPage(),
  onSubmit: reason =>
    dispatch(approvePause(arrearsId, pauseId, reason))
      .then(() => {
        redirectToDetailsPage();
        dispatch(
          addNotification({
            dataBddPrefix: 'approvePause',
            lines: reason
              ? null
              : [
                  { text: format(line1, { date: formatting.formatDate(activePause.endDate) }) },
                  { text: line2 },
                ],
            icon: 'success',
            notificationType: 'confirmation',
            title: reason ? cancelledTitle : approvedTitle,
          })
        );
      })
      .catch(() => {}),
  updatePageHeader: () => dispatch(updateRibbon({ title: heading, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
