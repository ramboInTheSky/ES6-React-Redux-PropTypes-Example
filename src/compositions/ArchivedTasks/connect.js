import { connect } from 'react-redux';
import path from 'ramda/src/path';

import getHelpers from '../../util/stateHelpers';
import { updateRibbon } from '../../ducks/ribbon';

export const mapStateToProps = (state, ownProps) => {
  const { getBreadcrumbs } = getHelpers(state);
  const {
    arrearsDetails: { arrearsDetailsLabels, heading: arrearsDetailHeading },
    archivedTasks: { breadcrumbs: pageTitle },
  } = state.dictionary;
  const arrearsId = path(['match', 'params', 'arrearsId'], ownProps);

  return {
    arrearsDetailsLabels,
    breadcrumb: getBreadcrumbs(
      pageTitle,
      {},
      {
        label: arrearsDetailHeading,
        to: `/arrears-details/${arrearsId}`,
      }
    ),
    arrearsDetailHeading,
    arrearsId,
    pageTitle,
  };
};

export const mergeProps = (
  { breadcrumb, arrearsDetailHeading, pageTitle, ...stateProps },
  { dispatch },
  ownProps
) => ({
  ...stateProps,
  arrearsDetailHeading,
  updatePageHeader: () => dispatch(updateRibbon({ title: pageTitle, breadcrumb })),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  null,
  mergeProps
);
