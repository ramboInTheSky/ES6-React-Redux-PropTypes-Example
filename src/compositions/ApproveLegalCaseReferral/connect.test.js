import * as actions from '../../ducks/ribbon';
import * as legalReferralActions from '../../ducks/legalReferral';
import * as notificationActions from '../../ducks/notifications';

import { mapStateToProps, mergeProps } from './connect';
import state from '../../../__mocks__/state';

jest.mock('../../util/downloadFile', () => (uri, name) => ({ uri, name }));

describe('ApproveLegalCaseReferral connector', () => {
  let mockState;
  let mapStateToPropsResult;
  let ownProps;

  beforeEach(() => {
    mockState = state();
    ownProps = {
      bar: 'baz',
      match: {
        params: {
          arrearsId: 'foo',
          submissionId: 'abc123',
        },
      },
    };
    mapStateToPropsResult = mapStateToProps(mockState, ownProps);
  });

  describe('mapStateToProps', () => {
    it('generates the correct props', () => {
      expect(mapStateToPropsResult).toMatchSnapshot();
    });
    it('generates the correct props for a housing manager', () => {
      mapStateToPropsResult = mapStateToProps(
        { ...mockState, user: { ...mockState.user, userType: 'manager' } },
        ownProps
      );
      expect(mapStateToPropsResult).toMatchSnapshot();
    });
  });

  describe('mergeProps', () => {
    let dispatchProps;
    let result;
    let redirectToDetailsPage;

    beforeEach(() => {
      dispatchProps = {
        dispatch: jest.fn(() => ({
          then: cb => {
            cb();
            return { catch: () => {} };
          },
        })),
      };
      actions.updateRibbon = jest.fn();
      legalReferralActions.cancelLegalReferral = jest.fn();
      legalReferralActions.getLegalReferral = jest.fn();
      legalReferralActions.setLegalReferralApproved = jest.fn();
      notificationActions.addNotification = jest.fn();
      redirectToDetailsPage = jest.fn();

      result = mergeProps(
        { ...mapStateToPropsResult, redirectToDetailsPage },
        dispatchProps,
        ownProps
      );
    });

    afterEach(() => {
      jest.clearAllMocks();
    });

    it('generates the correct props', () => {
      expect(result).toMatchSnapshot();
    });

    it('performs the correct actions for updatePageHeader', () => {
      result.updatePageHeader();
      expect(dispatchProps.dispatch).toHaveBeenCalled();
      expect(actions.updateRibbon).toHaveBeenCalled();
    });

    it('performs the correct actions for getLegalReferral', () => {
      result.getLegalReferral();
      expect(dispatchProps.dispatch).toHaveBeenCalled();
      expect(legalReferralActions.getLegalReferral).toHaveBeenCalledWith(
        mapStateToPropsResult.submissionId
      );
    });

    it('performs the correct actions for onBack', () => {
      result.onBack();
      expect(redirectToDetailsPage).toHaveBeenCalled();
    });

    it('performs the correct actions for downloadFile', () => {
      const file = result.downloadFile('abc123', 'foo');
      expect(file).toEqual({ uri: 'abc123', name: 'foo' });
    });

    describe('onSubmit', () => {
      it('performs the correct actions for an approval', () => {
        result.onSubmit();
        expect(dispatchProps.dispatch).toHaveBeenCalled();
        expect(legalReferralActions.setLegalReferralApproved).toHaveBeenCalledWith('abc123');
        const addNotification = notificationActions.addNotification.mock.calls[0][0];
        expect(notificationActions.addNotification).toHaveBeenCalled();
        expect(addNotification).toMatchSnapshot();
        expect(redirectToDetailsPage).toHaveBeenCalled();
      });

      it('performs the correct actions for onCancel', () => {
        result.onSubmit('Reasons');
        expect(dispatchProps.dispatch).toHaveBeenCalled();
        expect(legalReferralActions.cancelLegalReferral).toHaveBeenCalledWith('abc123', {
          noteText: 'Reasons',
        });
        const addNotification = notificationActions.addNotification.mock.calls[0][0];
        expect(notificationActions.addNotification).toHaveBeenCalled();
        expect(addNotification).toMatchSnapshot();
        expect(redirectToDetailsPage).toHaveBeenCalled();
      });
    });
  });
});
