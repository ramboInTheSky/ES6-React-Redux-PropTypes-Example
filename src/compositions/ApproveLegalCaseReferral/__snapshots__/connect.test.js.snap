// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApproveLegalCaseReferral connector mapStateToProps generates the correct props 1`] = `
Object {
  "activePause": Object {
    "approvalDetail": Object {
      "decisionDate": "2018-10-04T00:00:00Z",
      "isApproved": true,
      "userName": "system",
    },
    "cancelReason": "",
    "createdAt": "2018-10-04T15:32:32Z",
    "createdBy": "Stephanie Golder",
    "endDate": "2018-10-18T00:00:00Z",
    "furtherDetail": "This is a pause",
    "isExtended": false,
    "isManagerNotified": false,
    "lastModifiedAt": "2018-10-04T15:32:35Z",
    "lastModifiedBy": "SYSTEM",
    "notifcationDetail": Object {
      "notifiedUserName": "Hannah Ike",
      "wasNotified": false,
    },
    "originalStartDate": "2018-10-04T00:00:00Z",
    "owner": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "raisedBy": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "reasonId": "newbenefituc",
    "status": "Active",
  },
  "arrearsId": "foo",
  "breadcrumb": Array [
    Object {
      "href": "#{Dashboards.Core.CustomerBaseURL}",
      "id": "breadcrumb-dashboard",
      "label": "Dashboard",
    },
    Object {
      "id": "breadcrumb-arrearsDashboard",
      "label": "My arrears",
      "to": "/",
    },
    Object {
      "label": "Arrears details",
      "to": "/arrears-details/foo",
    },
    Object {
      "id": "breadcrumb-Review legal case referral",
      "label": "Review legal case referral",
    },
  ],
  "errorText": Object {
    "approveReferralError": "Please either approve or reject this legal referral",
    "formError": "There was a problem approving this legal referral, please try again.",
    "reasonForDenyingError": "Please give a reason for not approving this legal referral",
  },
  "formError": undefined,
  "heading": "Review legal case referral",
  "isManager": false,
  "labels": Object {
    "approveReferral": "Approve legal referral?",
    "approveReferralSubtitle": "You must review the legal referral pack, linked above, before you can answer this question",
    "backButton": "Back",
    "no": "no",
    "reasonForDenying": "Reason for not approving referral",
    "referralFormLink": "Legal referral form",
    "referralPackLink": "Legal referral pack",
    "submitButton": "Submit",
    "yes": "yes",
  },
  "loading": undefined,
  "notification": Object {
    "approvedTitle": "Legal case referral approved",
    "cancelledTitle": "Legal case referral cancelled",
  },
  "redirectToDetailsPage": [Function],
  "referralPack": undefined,
  "submissionId": "abc123",
  "subtitle": "Legal case awaiting housing manager approval",
}
`;

exports[`ApproveLegalCaseReferral connector mapStateToProps generates the correct props for a housing manager 1`] = `
Object {
  "activePause": Object {
    "approvalDetail": Object {
      "decisionDate": "2018-10-04T00:00:00Z",
      "isApproved": true,
      "userName": "system",
    },
    "cancelReason": "",
    "createdAt": "2018-10-04T15:32:32Z",
    "createdBy": "Stephanie Golder",
    "endDate": "2018-10-18T00:00:00Z",
    "furtherDetail": "This is a pause",
    "isExtended": false,
    "isManagerNotified": false,
    "lastModifiedAt": "2018-10-04T15:32:35Z",
    "lastModifiedBy": "SYSTEM",
    "notifcationDetail": Object {
      "notifiedUserName": "Hannah Ike",
      "wasNotified": false,
    },
    "originalStartDate": "2018-10-04T00:00:00Z",
    "owner": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "raisedBy": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "reasonId": "newbenefituc",
    "status": "Active",
  },
  "arrearsId": "foo",
  "breadcrumb": Array [
    Object {
      "href": "#{Dashboards.Core.CustomerBaseURL}",
      "id": "breadcrumb-dashboard",
      "label": "Dashboard",
    },
    Object {
      "id": "breadcrumb-arrearsDashboard",
      "label": "My arrears",
      "to": "/",
    },
    Object {
      "label": "Arrears details",
      "to": "/arrears-details/foo",
    },
    Object {
      "id": "breadcrumb-Review legal case referral",
      "label": "Review legal case referral",
    },
  ],
  "errorText": Object {
    "approveReferralError": "Please either approve or reject this legal referral",
    "formError": "There was a problem approving this legal referral, please try again.",
    "reasonForDenyingError": "Please give a reason for not approving this legal referral",
  },
  "formError": undefined,
  "heading": "Review legal case referral",
  "isManager": true,
  "labels": Object {
    "approveReferral": "Approve legal referral?",
    "approveReferralSubtitle": "You must review the legal referral pack, linked above, before you can answer this question",
    "backButton": "Back",
    "no": "no",
    "reasonForDenying": "Reason for not approving referral",
    "referralFormLink": "Legal referral form",
    "referralPackLink": "Legal referral pack",
    "submitButton": "Submit",
    "yes": "yes",
  },
  "loading": undefined,
  "notification": Object {
    "approvedTitle": "Legal case referral approved",
    "cancelledTitle": "Legal case referral cancelled",
  },
  "redirectToDetailsPage": [Function],
  "referralPack": undefined,
  "submissionId": "abc123",
  "subtitle": "Legal case awaiting housing manager approval",
}
`;

exports[`ApproveLegalCaseReferral connector mergeProps generates the correct props 1`] = `
Object {
  "activePause": Object {
    "approvalDetail": Object {
      "decisionDate": "2018-10-04T00:00:00Z",
      "isApproved": true,
      "userName": "system",
    },
    "cancelReason": "",
    "createdAt": "2018-10-04T15:32:32Z",
    "createdBy": "Stephanie Golder",
    "endDate": "2018-10-18T00:00:00Z",
    "furtherDetail": "This is a pause",
    "isExtended": false,
    "isManagerNotified": false,
    "lastModifiedAt": "2018-10-04T15:32:35Z",
    "lastModifiedBy": "SYSTEM",
    "notifcationDetail": Object {
      "notifiedUserName": "Hannah Ike",
      "wasNotified": false,
    },
    "originalStartDate": "2018-10-04T00:00:00Z",
    "owner": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "raisedBy": Object {
      "id": "039a3660-bd18-e811-80c8-005056825b41",
      "name": "Stephanie Golder",
      "type": "systemuser",
    },
    "reasonId": "newbenefituc",
    "status": "Active",
  },
  "arrearsId": "foo",
  "bar": "baz",
  "downloadFile": [Function],
  "errorText": Object {
    "approveReferralError": "Please either approve or reject this legal referral",
    "formError": "There was a problem approving this legal referral, please try again.",
    "reasonForDenyingError": "Please give a reason for not approving this legal referral",
  },
  "formError": undefined,
  "getLegalReferral": [Function],
  "heading": "Review legal case referral",
  "isManager": false,
  "labels": Object {
    "approveReferral": "Approve legal referral?",
    "approveReferralSubtitle": "You must review the legal referral pack, linked above, before you can answer this question",
    "backButton": "Back",
    "no": "no",
    "reasonForDenying": "Reason for not approving referral",
    "referralFormLink": "Legal referral form",
    "referralPackLink": "Legal referral pack",
    "submitButton": "Submit",
    "yes": "yes",
  },
  "loading": undefined,
  "match": Object {
    "params": Object {
      "arrearsId": "foo",
      "submissionId": "abc123",
    },
  },
  "onBack": [Function],
  "onSubmit": [Function],
  "referralPack": undefined,
  "subtitle": "Legal case awaiting housing manager approval",
  "updatePageHeader": [Function],
}
`;

exports[`ApproveLegalCaseReferral connector mergeProps onSubmit performs the correct actions for an approval 1`] = `
Object {
  "dataBddPrefix": "approveLegalReferral",
  "icon": "success",
  "notificationType": "confirmation",
  "title": "Legal case referral approved",
}
`;

exports[`ApproveLegalCaseReferral connector mergeProps onSubmit performs the correct actions for onCancel 1`] = `
Object {
  "dataBddPrefix": "approveLegalReferral",
  "icon": "success",
  "notificationType": "confirmation",
  "title": "Legal case referral cancelled",
}
`;
